tests/e2e/pipeline.go

```diff
@@ -488,21 +510,73 @@ func updatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expecte
 		Expect(err).ShouldNot(HaveOccurred())
 	}
 
+	Document("Verifying Pipeline got updated")
 	// get Pipeline to check that spec has been updated to correct spec
-	verifyPipelineSpec(Namespace, name, func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
-		return len(retrievedPipelineSpec.Vertices) == numPipelineVertices
-	})
-
-	verifyPipelineRolloutDeployed(name)
+	VerifyPipelineSpec(Namespace, name, verifySpecFunc)
```

Comment:
hmm, this one shouldn't pass if we're still in a long pause - is it?

Reply:
It is not meant to pass but I am just passing a function that returns `true` to circumvent that. Should we handle this another way?

Reply:
Oh, that's clever. But I do think it's a little hard to read and the comment makes it sound like it's "been updated to correct spec". I think it will be more apparent if we check for the expectedPhase being PipelinePhasePausing to determine whether this needs to be done. Also, maybe the function passed in should be a pointer since it should be optional if it's PipelinePhasePausing.